### PR TITLE
Add status to pull requests

### PR DESCRIPTION
**Pull request by pelson**
Closes https://github.com/conda-forge/conda-forge-webservices/pull/12

Takes #12 further to abstract more information out of the compute_lint_message function.

### CODE REVIEW

**Comment by jakirkham** [comment]
Sorry, I haven't played with the status message for awhile. Thanks for giving it a try. If there is anything I can do to help, please let me know.

**Comment by pelson** [comment]
> Sorry, I haven't played with the status message for awhile. Thanks for giving it a try. If there is anything I can do to help, please let me know.

No problems. I'm just trying to pull some of the context out - you were right that all of the bits were conveniently in that function, but I didn't want to pollute its purpose.
Hopefully this change will take it in a healthy direction.

**Comment by jakirkham** [comment]
> No problems. I'm just trying to pull some of the context out - you were right that all of the bits were conveniently in that function, but I didn't want to pollute its purpose.

Yeah, it was a bit tricky, but I think we could figure something out. We just have to balance speed/time with quality here. Also, we can always go back and refactor.

> Hopefully this change will take it in a healthy direction.

Sure, it has tests. So, that is already an improvement on what I did. :smile:

File: conda_forge_webservices/linting.py [hunk 1]
```diff
@@ -1,22 +1,14 @@
-# TODO: Add an interface to do this from the CLI:
-# conda-linting-service conda-forge/staged-recipes 123
-
+from contextlib import contextmanager
+from glob import glob
 import os
-import tornado.escape
-import tornado.httpserver
-import tornado.ioloop
-import tornado.web
```
Comment (jakirkham) [inline conda_forge_webservices/linting.py]: Were we just never using these?

Comment (pelson) [inline conda_forge_webservices/linting.py]: Nope. They were split out into the webapp code.

**Comment by pelson** [comment]
So I think this is good to go, but I'm not around now until 12hrs, so will hold fire on deploying this until then.

File: conda_forge_webservices/linting.py (hunk 1 above)
Comment (jakirkham) [inline conda_forge_webservices/linting.py]: Ah ok. So these are just leftover imports. Thanks for clarifying.

**Comment by jakirkham** [comment]
Is this failure just caused by having it on your fork? Should we move the PR over to the main repo?

**Comment by jakirkham** [comment]
Here is what I am seeing with the status.

* Show up in the failing case. :+1:
* Missing on the passing. (should be fixed)
* Missing on the no recipe case. (should it show up?)

Seems like the passing case needs some help.

As for the missing case, not sure what to do. Should we pass it and say no recipes found in the summary? Just simply not have a status (as now)? Some other behavior? I think it would be nice to have some status just to verify it is working, but it probably doesn't need to be the same as a good recipe. Though it should not be a failure.

**Comment by pelson** [comment]
> Is this failure just caused by having it on your fork?

Yes. I'm able to run these locally using my own PAT:

```
GH_TOKEN=$(cat ~/.conda-smithy/github.token.pelson) nosetests conda_forge_webservices
```

You should be able to run the tests locally too with your own if you like.

**Comment by pelson** [comment]
Deployed. I've upgraded the dyno to hobby and tested this with https://github.com/conda-forge/staged-recipes/pull/364 (and we are getting a nice status!) 🎉

**Comment by jakirkham** [comment]
Thanks @pelson for bringing this one home. This is truly wonderful.

**Comment by jakirkham** [comment]
So, this one ( https://github.com/conda-forge/staged-recipes/pull/365 ) appears to be passing, but the recipes still have no tests.

**Comment by pelson** [comment]
> So, this one ( conda-forge/staged-recipes#365 ) appears to be passing, but the recipes still have no tests.

Is that just because it hasn't triggered a lint since it was deployed? Looks like the last check was > 7 hours ago...

**Comment by jakirkham** [comment]
Sorry, I was going to mention that soon after it appeared correct. Either I'm not awake :sleeping: or something magical happened :fireworks:. The failure is showing up correctly now.

**Pull request merged**